src/include/sof/lib_manager.h

```diff
@@ -0,0 +1,204 @@
+ *   |                       |                            | Prepare Storage Memory   |         |
+ *   |                       |                            | -------                  |         |
+ *   |                       |                            |        |                 |         |
+ *   |                       |                            | <------                  |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            | Copy Library Data        |         |
+ *   |                       |                            | -----------------------> |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |  Verify Manifest         |         |
+ *   |                       |                            | -------------------------|-------> |
+ *   |                       |                            |  results                 |         |
+ *   |                       |                            | <------------------------|-------- |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            | Update Library           |         |
+ *   |                       |                            | descriptors table        |         |
+ *   |                       |                            | -------                  |         |
+ *   |                       |                            |        |                 |         |
+ *   |                       |                            | <------                  |         |
+ *   |                       |       return status        |                          |         |
+ *   |                       | <------------------------- |                          |         |
+ *   | Complete IPC request  |                            |                          |         |
+ *   | <-------------------  |                            |                          |         |
+ *
+ * Driver                   IPC4                       Library                     MEMORY     SHA
+ *                         Handler                     Manager                               Engine
+ */
+
+#ifndef __SOF_LIB_MANAGER_H__
+#define __SOF_LIB_MANAGER_H__
+
+#include <stdint.h>
+#include <rimage/cavs/cavs_ext_manifest.h>
+#include <rimage/sof/user/manifest.h>
+
+#define MAX_LIBS				16
+#define LIB_ID_SHIFT			12
+#define LIB_NOTIX_MAX_COUNT		4
+
+#define GET_LIB_ID(module_id) (module_id >> LIB_ID_SHIFT)
+#define GET_MODULE_INDEX(module_id) (module_id & 0xFFF)
+
```

Comment:
Use `LIB` and `lib` prefixes wherever possible. e.g `ipc_lib_msg` -> `lib_ipc_msg`.